Core/clim-core/commands.lisp

```diff
@@ -1294,33 +1294,54 @@ examine the type of the command menu item to see if it is
   (with-presentation-type-parameters (command maybe-supertype)
     (let ((super-table command-table))
       (with-presentation-type-parameters (command type)
-	(command-table-inherits-from-p super-table command-table)))))
+        (command-table-inherits-from-p super-table command-table)))))
 
 (define-presentation-method present (object (type command)
-				     stream
-				     (view textual-view)
-				     &key acceptably for-context-type)
+                                     stream
+                                     (view textual-view)
+                                     &key acceptably for-context-type)
   (declare (ignore acceptably for-context-type))
   (funcall *command-unparser* command-table stream object))
 
 (define-presentation-method accept ((type command) stream
-				    (view textual-view)
-				    &key)
-  (let ((command (funcall *command-parser* command-table stream)))
-    (cond ((null command)
-	   (simple-parse-error "Empty command"))
-          ((partial-command-p command)
-           (funcall *partial-command-parser*
-            command-table stream command
-            (position *unsupplied-argument-marker* command)))
-	  (t (values command type)))))
+                                    (view textual-view)
+                                    &key)
+  (setq command-table (find-command-table command-table))
+  (let ((start-position (and (input-editing-stream-p stream)
+                             (stream-scan-pointer stream)))
+        (replace-input-p nil))
+    (multiple-value-bind (object new-type)
+        ;; We establish a new input context so that clicks on command-menu-items
+        ;; throw to us. This will let's us handle "partial commands" below.
+        ;; This only works for partial commands coming by clicking on
+        ;; application-frame menus.  Clicking on a presentation of something as a
+        ;; command with unsupplied arguments with instead by caught by accept-1
+        ;; and bypass this handling.
+        (with-input-context ('command :override nil)
+            (object presentation-type event options)
+            (funcall *command-parser* command-table stream)
+          (t (when (getf options :echo t)
+               (setq replace-input-p t))
+             (values object presentation-type)))
+      (declare (ignore new-type))
+      (if (partial-command-p object)
+          (values (funcall *partial-command-parser*
+                           command-table stream object start-position)
+                  type)
+          (progn (when replace-input-p
+                   (presentation-replace-input stream object type view
+                                               :buffer-start start-position
+                                               ;; This is important in case the clicked
+                                               ;; on thing returned something unparseable
+                                               :rescan nil))
+                 (values object type))))))
```

Comment:
is this code written from scratch? It resembles the code from clim-tos[1] and it is fine, because licenses are compatible, but in that case we should add

```
;;;"Copyright (c) 1990, 1991, 1992 Symbolics, Inc.  All rights reserved.
;;; Portions copyright (c) 1989, 1990 International Lisp Associates."
```
to our own copyrights in this file. Looking at the latter (clim-tos code) I tend to agree, that input context should be the type variable, because said variable may be in fact `(command :command-table foo)`, and it is more accurate.

https://gitlab.common-lisp.net/mcclim/gramps-clim2/-/blob/master/clim/command-processor.lisp#L611